lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -44,6 +45,11 @@
         description:
             - Name of the resource group containing the virtual machine.
         required: true
+    resource:
+        description:
+            - The base URL for the Resource Manager API endpoint (e.g. https://management.azure.com)
+        required: false
```

Comment:
`required: false` is redundant- doesn't hurt anything, but doesn't need to be there...